project/src/main/java/com/max/grpc/project/EchoServer.java

```diff
@@ -0,0 +1,62 @@
+
+package com.max.grpc.project;
+
+import com.max.grpc.protos.EchoEntity;
+import com.max.grpc.protos.EchoServiceGrpc;
+
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.stub.StreamObserver;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+public class EchoServer {
+    private static final Logger logger = Logger.getLogger(EchoServer.class.getName());
+
+    private int port = 50000;
+    private Server server;
+
+    public void start() throws IOException {
+        server = ServerBuilder.forPort(port)
+                .addService(new EchoImpl())
+                .build()
+                .start();
+        logger.info("Server started, listening on " + port);
+    }
+
+    public void stop() throws InterruptedException {
+        if (server != null) {
+            server.shutdown().awaitTermination(30, TimeUnit.SECONDS);
+        }
+    }
+
+    public void blockUntilShutdown() throws InterruptedException {
+        if (server != null) {
+            server.awaitTermination();
+        }
+    }
+
+    public static void main(String[] args) throws IOException, InterruptedException {

```

Comment:
move out main from the server class. It's Ok only for samples but you have to have real project

Reply:
Already moved